File: Partie-Backend/src/main/java/com/basketapp/basket_app/models/Utilisateur.java
package com.basketapp.basket_app.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinTable;
import javax.persistence.Lob;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data 
@NoArgsConstructor 
@AllArgsConstructor
@Entity
public class Utilisateur {
    @Id
    @GeneratedValue (strategy=GenerationType.IDENTITY)
    private Long idUtil;

    @Lob
    String image;

    @Column(unique=true)
    private String username;
    private String password;
    private Boolean enabled;
    private String nomUtil;
	private String prenomUtil;
	private String numTelUtil;
	private String emailUtil;

    @ManyToMany(cascade=CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinTable(name="utilisateur_role",joinColumns = @JoinColumn(name="idUtil") ,
    inverseJoinColumns = @JoinColumn(name="idRole"))
	private List<Role> roles;
}

File: Partie-Mobile/lib/models/Utilisateur.dart
class Utilisateur {
  int? idUtil;
  String? image;
  String? username;
  String? password;

  String? nomUtil;
  String? prenomUtil;
  String? numTelUtil;
  String? emailUtil;
  List<Roles>? roles;

  Utilisateur(
      {this.idUtil,
       this.image,
      this.username,
      this.password,
     
      this.nomUtil,
      this.prenomUtil,
      this.numTelUtil,
      this.emailUtil,
      this.roles});

  Utilisateur.fromJson(Map<String, dynamic> json) {
    idUtil = json['idUtil'];
    image = json['image'];
    username = json['username'];
    password = json['password'];
    
    nomUtil = json['nomUtil'];
    prenomUtil = json['prenomUtil'];
    numTelUtil = json['numTelUtil'];
    emailUtil = json['emailUtil'];
    if (json['roles'] != null) {
      roles = <Roles>[];
      json['roles'].forEach((v) {
        roles!.add(new Roles.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idUtil'] = this.idUtil;
    data['image'] = this.image;
    data['username'] = this.username;
    data['password'] = this.password;
 
    data['nomUtil'] = this.nomUtil;
    data['prenomUtil'] = this.prenomUtil;
    data['numTelUtil'] = this.numTelUtil;
    data['emailUtil'] = this.emailUtil;
    if (this.roles != null) {
      data['roles'] = this.roles!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class Roles {
  int? idRole;
  String? role;

  Roles({this.idRole, this.role});

  Roles.fromJson(Map<String, dynamic> json) {
    idRole = json['idRole'];
    role = json['role'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idRole'] = this.idRole;
    data['role'] = this.role;
    return data;
  }
}